fix(docs): Add workflows and tox envs for building docs

Carsons-Eels opened the pull request:
## 🗒️ Description

WELDs the documentation infrastructure from [EEST](https://github.com/ethereum/execution-spec-tests/) into the project including automated workflows, tox environments, and tooling to maintain documentation quality.

### Key Changes

**GitHub Workflows:**
- Added `.github/workflows/test-docs.yaml` workflow to automatically build and validate EEST documentation on push and pull request events
  - Workflow triggers on changes to `packages/tests/**`, `tests/**`, and the workflow file itself

**Tox Environments:**
- `changelog`: Validates changelog entries
- `mkdocs`: Builds HTML documentation using mkdocs with strict mode enabled
- `markdownlint`: Lints markdown files using markdownlint
- `spec-docs`:  Generate documentation for the specification code using docc

**Documentation Tooling:**
- Added `src/ethereum_spec_tools/whitelist.py`: Python script to manage the codespell whitelist with proper alphabetical sorting and formatting
- Added EESTs `.markdownlint.yaml`: Configuration for markdownlint with appropriate rule customizations
- Updated `whitelist.txt`: Consolidated EEST's whitelist into ours, cleaned up codespell whitelist entries 
- Added `docs/templates/function_parameter_datatable.html.j2`: Jinja2 template for rendering function parameter documentation

**Configuration Updates:**
- Updated `tox.ini` with new documentation environments and proper dependencies
- Added mkdocs-related dependencies (`mkdocstrings[python]`, etc.) to documentation extras
- Configured proper path triggers for CI workflows

### Benefits

1. **Automated Quality Checks**: Documentation is automatically validated on every PR
2. **Consistent Formatting**: Standardized markdown linting and spell checking
3. **Developer Experience**: Easy-to-use tox commands for local documentation building and validation
4. **Maintainability**: Centralized whitelist management tool prevents ad-hoc additions

### Usage

```bash
# Build EEST documentation
uv run tox -e eest-mkdocs

# Run all EEST doc checks
uv run tox -e eest-docs

# Run all documentation checks (EEST + main specs)
uv run tox -e docs

# Add words to codespell whitelist
uv run whitelist word1 word2...
```

## 🔗 Related Issues or PRs

Part of the WELD milestone (❄️ WELD Freeze ❄️)

## ✅ Checklist

- [x] All: Ran fast `tox` checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    uvx --with=tox-uv tox -e static
    ```
- [x] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope):`.
- [x] All: Considered adding an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] All: Set appropriate labels for the changes (only maintainers can apply labels).
- [x] Tests: Ran `mkdocs serve` locally ~~and verified the auto-generated docs for new tests in the [Test Case Reference](https://eest.ethereum.org/main/tests/) are correctly formatted.~~
- [N/A ] ~~Tests: For PRs implementing a missed test case, update the [post-mortem document](/ethereum/execution-spec-tests/blob/main/docs/writing_tests/post_mortems.md) to add an entry the list.~~
- [N/A ] ~~Ported Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) or [tests/static](/ethereum/execution-spec-tests/blob/main/tests/static) have been assigned `@ported_from` marker.~~
- [x] Find-replace mkdocstrings prefixes: `:::` with the new package paths
- [x] Finish updating new whitelist tool to be installable and directly invokable (e.g. `whitelist word1 word2`)

#### Cute Animal Picture

![Put a link to a cute animal picture inside the parenthesis-->](https://www.publicdomainpictures.net/pictures/590000/velka/image-17101872434zD.jpg)
Credit to Linnaea Mallette: 
- :spider_web: [website](http://linnaeamallette.com/)
- :book: [Facebook](https://www.facebook.com/Linnaea-Mallette-Author-Speaker-Photographer-2411455605746608)
- :coin: [Tip the Artist!](https://www.paypal.com/donate?token=Q2O9ZVLC3GK2H1vYFvtN79YjtxUG_Q9ml9tetNUl0H30uqhfBjjuZ5sdI0dw9gIYvDPQNJeE58zD-M7e&locale.x=US)
License: [CC0 Public Domain](http://creativecommons.org/publicdomain/zero/1.0/)

danceratopz committed:
fix(tests): fix-up path in checklist plugin post-weld

Carsons-Eels committed:
chore: add workflows and tox envs for building docs

marioevz commented on .github/workflows/eest-docs.yaml:
```diff
@@ -0,0 +1,50 @@
+      - 'forks/**'
+    paths:
+      - 'src/ethereum_spec_tests/**'
+      - 'tests/eest/**'
```
```suggestion
      - 'tests/**'
```

Carsons-Eels replied:
Missed this one, good catch

marioevz commented on .github/workflows/eest-docs.yaml:
```diff
@@ -0,0 +1,50 @@
+      - mainnet
+      - 'forks/**'
+    paths:
+      - 'src/ethereum_spec_tests/**'
```
```suggestion
      - 'packages/tests/**'
```

or
```python
      - 'packages/testing/**'
```
if we decide to change it.

spencer-tb replied:
It changed!

marioevz commented on .github/workflows/eest-docs.yaml:
I think we should start renaming this to `testing-docs.yaml`, wdyt?

danceratopz replied:
Tbh, these will be the landing page for all of executions-specs docs, so just docs is more appropriate?

Carsons-Eels replied:
I think for now we could leave it as-is since we are planing to do an overhaul and unification of the documentation after WELDing is complete, then just go with `docs.yaml`

marioevz commented on .github/workflows/eest-docs.yaml:
```diff
@@ -0,0 +1,50 @@
+    paths:
+      - 'src/ethereum_spec_tests/**'
+      - 'tests/eest/**'
+      - '.github/workflows/eest-docs.yaml'
```
```suggestion
      - '.github/workflows/testing-docs.yaml'
```
If the comment to the file name is applied.

marioevz commented on .github/workflows/eest-docs.yaml:
```diff
@@ -0,0 +1,50 @@
+    paths:
+      - 'src/ethereum_spec_tests/**'
+      - 'tests/eest/**'
+      - '.github/workflows/eest-docs.yaml'
```
To be changed too if the previous comments are applied.

marioevz commented on tox.ini:
```diff
@@ -4,6 +4,15 @@ requires =
+runner = uv-venv-lock-runner
+package = editable
+wheel_build_env = .pkg
+python_source_dirs = src/ethereum_spec_tests tests # .github/scripts
```
```suggestion
python_source_dirs = packages/tests tests # .github/scripts
```

or
```python
python_source_dirs = packages/testing tests # .github/scripts
```
if we decide to change it.

spencer-tb replied:
It changed!

marioevz commented on tox.ini:
```diff
@@ -4,6 +4,15 @@ requires =
+wheel_build_env = .pkg
+python_source_dirs = src/ethereum_spec_tests tests # .github/scripts
+ruff_config = ruff.eest.toml
+mypy_config = mypy.eest.ini
```
I think these don't exist anymore.

marioevz reviewed (commented):
Gave this a quick look and left some comments, thanks!

danceratopz commented on .markdownlint.yaml:
```diff
@@ -0,0 +1,10 @@
+# All rules: https://github.com/DavidAnson/markdownlint#rules--aliases
+default: true # Default state for all rules
+MD013: false # line-length: We don't fill paragaraphs/limit line length
```
fix typo )))
```suggestion
MD013: false # line-length: We don't fill paragraphs/limit line length
```

Carsons-Eels commented:
After #1654 is merged, will need to find-replace `mkdocstrings` prefixes: `:::`  with the new package paths

Carsons-Eels committed:
chore: fix typo

Co-authored-by: danceratopz <danceratopz@gmail.com>

Carsons-Eels committed:
chore: ignore site dir for docs build, fix path issue

Carsons-Eels committed:
chore: remove old configs from tox

spencer-tb commented on .github/workflows/eest-docs.yaml:
```diff
@@ -0,0 +1,50 @@
+    paths:
+      - 'src/ethereum_spec_tests/**'
+      - 'tests/**'
+      - '.github/workflows/eest-docs.yaml'
```
Should we add the `docs/` as a path here as well, or a todo to add them?

spencer-tb commented on tox.ini:
```diff
@@ -121,3 +127,60 @@ extras = doc
+    GITHUB_ACTIONS
+    GITHUB_STEP_SUMMARY
+commands =
+    codespell --ignore-words=whitelist-eest.txt {[eest]python_source_dirs} docs
```
I think we should merge our whitelist with with yours. And then add the docs as path to the static spellcheck tox environment.

spencer-tb replied:
Ahh maybe skip the last part, I see the docs env below!

Carsons-Eels replied:
Reworked it a bunch, check the new version and let me know what you think. Thanks!

spencer-tb reviewed (commented):
Thanks! Some smol comments. Can re-evaluate once the weld has progressed.
Maybe we should change `eest` to `ethereum-testing` or `testing` @danceratopz

Carsons-Eels committed:
chore: fix rest of added eest tox jobs

Carsons-Eels committed:
chore: consolidate spellcheck

Carsons-Eels committed:
chore: add whitelist tool to tox

SamWilsn commented on docs/templates/function_parameter_datatable.html.j2:
```diff
@@ -42,14 +42,14 @@
 </div>
 <table id="test_table" class="display compact nowrap">
-    <thead>
+    <thread>
```
Huh? `thead` is a tag, `thread` is not.

Carsons-Eels replied:
:facepalm: I was tired when I was fixing the spelling stuff. Will fix!

SamWilsn replied:
Still says `thread` on my end.

danceratopz replied:
Fixed.

SamWilsn commented on scripts/add_to_whitelist.py:
Don't include the `.py` in the same of scripts meant to be executed directly. Just use a shebang (eg. `#!/usr/bin/env python3`).

If you ever change the implementation language, you'll have to change the file name and break people's workflows.

SamWilsn commented on scripts/add_to_whitelist.py:
Should this be a `src/ethereum_spec_tool`?

Carsons-Eels replied:
Yeah, I was talking to Mario about that. I threw it in tox just to get it up there quickly and test it out, but had planned on making it a tool you don't have to `uvx --with=tox-uv tox -e ...` every time because that's super annoying

Carsons-Eels replied:
Moved it over there, and started trying to set it up to be installable and invokable like `ethereum-spec-lint`, but haven't quite managed that yet. Might need a hand sorting that out.

SamWilsn commented on scripts/add_to_whitelist.py:
```diff
@@ -0,0 +1,61 @@
+        print("Example: ./whitelist_manager.py foo bar baz")
+        sys.exit(1)
+
+    whitelist_file = 'whitelist.txt'
```
This will be relative to the current working directory, so something like the following would give unexpected results:

```bash
$ cd scripts
$ ./add_to_whitelist foobar
```

Carsons-Eels replied:
Okay, I fixed it so that it searches upwards from the scripts location for the root `pyproject.toml` and then uses that path to access the whitelist. Tested and working.

SamWilsn commented on scripts/add_to_whitelist.py:
```diff
@@ -0,0 +1,61 @@
+
+    if not new_words:
+        print("Usage: ./whitelist_manager.py <word1> <word2> <word3> ...")
+        print("Example: ./whitelist_manager.py foo bar baz")
```
These don't match the script file name. How would you feel about optparse or argparse?

Carsons-Eels replied:
I'll upgrade it, yeah. I just did it the fast way to get the first version hacked out

SamWilsn commented on scripts/add_to_whitelist.py:
```diff
@@ -0,0 +1,61 @@
+
+    # Read existing whitelist (create empty list if file doesn't exist)
+    existing_words = []
+    if os.path.exists(whitelist_file):
```
This pattern creates a [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) race. Not a huge problem here, but it's better to just use `open(..., "r+")` directly.

SamWilsn commented on scripts/add_to_whitelist.py:
```diff
@@ -0,0 +1,61 @@
+    print(f"Total unique entries: {len(all_words)}")
+
+    # Sort alphabetically (case-insensitive)
+    sorted_words = sorted(all_words, key=str.lower)
```
You should use [`str.casefold`](https://docs.python.org/3/library/stdtypes.html#str.casefold) whenever doing case-insensitive comparisons in Python.

SamWilsn commented on scripts/add_to_whitelist.py:
```diff
@@ -0,0 +1,61 @@
+        current_letter = word[0].lower()
+
+        # If we're starting a new letter (and it's not the first word), add a blank line
+        if current_letter != previous_letter and len(words_with_separators) > 0:
```
```suggestion
        if current_letter != previous_letter and words_with_separators:
```

SamWilsn commented on tox.ini:
```diff
@@ -4,6 +4,12 @@ requires =
+runner = uv-venv-lock-runner
+uv_seed = false
+wheel_build_env = .pkg
+python_source_dirs = src/ethereum_spec_tests tests packages/tests .github/scripts
```
What does this env do?

Carsons-Eels replied:
leftovers from porting, I just forgot to get rid of it, but it's gone now

Carsons-Eels committed:
chore: fix markdownlint

Carsons-Eels committed:
fix: fix paths for doc generation

Carsons-Eels committed:
refactor: change import to avoid noqa

danceratopz committed:
chore(docs): rename files under docs/library/

danceratopz committed:
fix(docs): fix imports in gen_test_case_reference

danceratopz committed:
fix(docs): update mkdocstrings ::: directives for package renaming

danceratopz committed:
chore: update package links and code for package renaming

danceratopz committed:
chore: more package and source path fixes for package renaming

Carsons-Eels committed:
chore: migrate whitelist to ethereum_spec_tools; update md and tox

Carsons-Eels committed:
chore: fix paths in workflows, move whitelist to ethereum_spec_tools

Carsons-Eels committed:
chore: fix naming of workflow

Carsons-Eels committed:
chore: final whitelist update

Carsons-Eels committed:
chore: upgrade whitelist to add whitelist file location detection

Carsons-Eels committed:
chore: fix mypy errors in whitelist script

Carsons-Eels committed:
chore: fix changelog tool

Carsons-Eels committed:
chore: fix markdownlint

Carsons-Eels commented:
It was all working, then I rebased onto the latest forks/osaka and it broke again :woman_facepalming: 
Fixed changelog and markdownlint (bit hacky, but it works) but mkdocs is more of a frustration to fix. Tried moving the docs folder and mkdocs.yml into ./packages/testing but still can't get it to gel properly. 
Will try to sort it out tomorrow.

danceratopz commented:
> It was all working, then I rebased onto the latest forks/osaka and it broke again 🤦‍♀️ Fixed changelog and markdownlint (bit hacky, but it works) but mkdocs is more of a frustration to fix. Tried moving the docs folder and mkdocs.yml into ./packages/testing but still can't get it to gel properly. Will try to sort it out tomorrow.

As I understood, `mkdocs.yml` and `./docs` should stay in our repo root as it will contain the documentation for the whole repository; after updating they will no longer be tied to EEST (or `./packages/testing/`).

Carsons-Eels commented:
Hmmm, yeah that makes sense. Trying to move it turned into a bigger mess anyway

Carsons-Eels committed:
chore: fix mkdocs

SamWilsn commented on .github/workflows/test-docs.yaml:
```diff
@@ -0,0 +1,52 @@
+name: Spec Docs
```
Spec Docs?

danceratopz replied:
Updated.

SamWilsn commented on .github/workflows/test-docs.yaml:
```diff
@@ -0,0 +1,52 @@
+      - 'packages/testing/**'
+      - 'tests/**'
+      - '.github/workflows/test-docs.yaml'
+      - 'docs/**'
```
Why no `uv.lock`?

SamWilsn commented on .github/workflows/test-docs.yaml:
```diff
@@ -0,0 +1,52 @@
+      - 'packages/testing/**'
+      - 'tests/**'
+      - '.github/workflows/test-docs.yaml'
+      - 'docs/**'
```
Could use a YAML anchor (see `gh-pages.yml`) for an example.

Carsons-Eels committed:
chore: update gh-pages workflow to reference new tox env for docc

SamWilsn commented on EIP_AUTHORS_MANUAL.md:
```diff
@@ -1,12 +1,14 @@
-# Introduction
+# EIP AUTHOR MANUAL
```
```suggestion
# EIP Author's Manual
```

SamWilsn commented on docs/scripts/gen_test_case_reference.py:
```diff
@@ -58,19 +66,24 @@
 logger.info(
-    f"Generating documentation for test cases until {GENERATE_UNTIL_FORK} as fill {' '.join(args)}"
+    f"Generating documentation for test cases until {GENERATE_UNTIL_FORK} as \
+    fill {' '.join(args)}"
```
```suggestion
    f"Generating documentation for test cases until {GENERATE_UNTIL_FORK} as "
    f"fill {' '.join(args)}"
```

Adjacent strings concatenate in Python as has less murky whitespace handling.

SamWilsn commented on docs/scripts/generate_fill_help.py:
```diff
@@ -98,10 +104,15 @@ def generate_command_line_options_docs():
-    logger.info("Generated filling_tests_command_line_options.md with current fill --help output")
+    logger.info(
+        "Generated filling_tests_command_line_options.md with current fill \
+            --help output"
```
Similar comment about adjacent strings.

danceratopz replied:
Fixed.

SamWilsn commented on mkdocs.yml:
```diff
@@ -0,0 +1,137 @@
+  - literate-nav:
+      nav_file: navigation.md
+
+watch:
+  - CONTRIBUTING.md
+  - SECURITY.md
+  - src/
+  - tests/
```
Should `packages` be in here, or does it only rely on `tests`?

danceratopz replied:
Good catch, it does require `./tests` to generate the "Test Case Reference", but should require `./packages/testing/src/` (instead of `./src/`) to pull out various docstrings auto-included in the docs and generate the testing library reference. 

Tbh, the `watch` feature doesn't work very well though.

danceratopz replied:
Fixed, and updated other EEST specific config here.

SamWilsn commented on mkdocs.yml:
```diff
@@ -0,0 +1,137 @@
+
+extra:
+  version:
+    provider: mike
```
Mike's doing the lord's work.

danceratopz replied:
Not for much longer if he can't avoid bloating our git history! :sweat_smile:

SamWilsn commented on packages/testing/src/execution_testing/cli/pytest_commands/plugins/filler/eip_checklist.py:
```diff
@@ -51,12 +51,8 @@ def pytest_addoption(parser: pytest.Parser) -> None:
-    / "checklist_templates"
-    / "eip_testing_checklist_template.md"
+TEMPLATE_PATH = Path(
+    "docs/writing_tests/checklist_templates/eip_testing_checklist_template.md"
```
The meaning has changed here, is that intentional? Originally this used the path of the current file, while the new version uses the current working directory.

danceratopz replied:
Thanks! I reverted this back to the original meaning. This change would have broken eip checklist generation if it was not ran from the repo root.

SamWilsn commented on packages/testing/src/execution_testing/cli/tox_helpers.py:
```diff
@@ -55,6 +55,17 @@ def write_github_summary(
+    """Locate the root directory of this project."""
+    # Search upwards from file location
+    script_dir = Path(__file__).resolve().parent
+    for parent in [script_dir, *script_dir.parents]:
```
Perhaps I'm naive but would `script_dir.parents[1:]` (or similar) work here?

danceratopz replied:
I think this is ok.

SamWilsn commented on pyproject.toml:
```diff
@@ -396,6 +424,7 @@ exclude = [
     "^fixtures/",
     "^logs/",
     "^site/",
+    "^tests/json_infra/"
```
Why is this being excluded?

danceratopz replied:
Added back.

SamWilsn commented on tox.ini:
```diff
@@ -100,13 +100,55 @@ commands =
+dependency_groups = docs
+changedir = {toxinidir}/packages/testing
+allowlist_externals = uv
+commands_pre = uv pip install -e {toxinidir}/packages/testing
```
Why is this expressed as a command and not as a `deps`?

SamWilsn commented on tox.ini:
```diff
@@ -100,13 +100,55 @@ commands =
+commands_pre = uv pip install -e {toxinidir}/packages/testing
+commands =
+    ruff check --no-fix --show-fixes docs/scripts/
+    ruff format --check docs/scripts/
```
Should these not be part of `static`?

SamWilsn commented on tox.ini:
```diff
@@ -100,13 +100,55 @@ commands =
+description = Validate EEST changelog entries
+dependency_groups = docs
+changedir = {toxinidir}/packages/testing
+allowlist_externals = uv
```
This makes me uncomfortable. Isn't `uv` installed by `tox-uv`?

danceratopz committed:
refactor: use string concatenation instead of continuation

Co-authored-by: Sam Wilson <57262657+SamWilsn@users.noreply.github.com>

danceratopz committed:
fix: fix-up tag in j2 template

danceratopz committed:
refactor: use string concatenation instead of continuation

danceratopz committed:
chore: remove /docs from gitignore

danceratopz commented on scripts/add_to_whitelist.py:
This file can be removed; the script now lives in `src/ethereum_spec_tools/whitelist/`.

danceratopz commented on src/ethereum_spec_tools/whitelist/__init__.py:
```diff
@@ -0,0 +1,124 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
```
I think a copyright notice here is overkill. It's only added to `docc`. The repo root sets CC0.

danceratopz replied:
Removed.

danceratopz committed:
chore: update mkdocs config for the weld

danceratopz commented on src/ethereum_spec_tools/whitelist/__init__.py:
```diff
@@ -0,0 +1,124 @@
+            print(f"Total unique entries: {len(all_words)}")
+
+        # Sort alphabetically
+        sorted_words = sorted(all_words, key=str.casefold)
```
This doesn't appear to order case-sensitive variants of the same word deterministically and leads to a re-ordering of the file when re-ran, creating a large diff.

danceratopz replied:
Fixed.

danceratopz committed:
chore: remove license from whitelist

danceratopz committed:
fix: ensure case-sensitive words are ordered deterministically

danceratopz committed:
chore: apply determistic whitelist ordering

danceratopz committed:
chore: add thead for jinja2 templates to whitelist

danceratopz committed:
chore: remove duplicate whitelist script

danceratopz reviewed (commented):
Few comments on whitelist.

danceratopz committed:
chore(docs): improve the title of eip authors document

Co-authored-by: Sam Wilson <57262657+SamWilsn@users.noreply.github.com>

danceratopz commented on pyproject.toml:
```diff
@@ -364,10 +394,8 @@ skip = [                        # Don't check these files/folders
     "*.coverage*",
     "__pycache__",
     ".ruff_cache",
-    "fixtures.*",
```
I think we can keep this.

danceratopz committed:
chore: add back paths to codespell & mypy config

danceratopz committed:
chore: specify codespell ignore-words in pyproject.toml

danceratopz commented on packages/testing/src/execution_testing/cli/tox_helpers.py:
```diff
@@ -210,7 +226,8 @@ def validate_changelog() -> None:
     """
-    changelog_path = Path("docs/CHANGELOG.md")
+    project_root = find_project_root()
+    changelog_path = Path(project_root / "docs/CHANGELOG.md")
```
`uv run markdownlintcli2_soft_fail` only works from the project root (or it won't find its config file `.markdownlint.yaml`, so we should rather just check that a `pyproject.toml` is present in the cwd and give an appropriate error message if not.

danceratopz commented on packages/testing/src/execution_testing/cli/tox_helpers.py:
```diff
@@ -55,6 +55,17 @@ def write_github_summary(
+    # Search upwards from file location
+    script_dir = Path(__file__).resolve().parent
+    for parent in [script_dir, *script_dir.parents]:
+        if (parent / "EIP_AUTHORS_MANUAL.md").exists():
```
We should find a more dependable way to detect the project root than `EIP_AUTHORS_MANUAL.md` (this might be removed). `pyproject.toml` alone won't work as we now have multiple `pyproject.toml`s (in the repo root and in `packages/`). Will update to search for `.git` dir and `pyproject.toml`.

danceratopz committed:
refactor: don't use markdown file to detect repo root

danceratopz committed:
chore: remove pydantic from lint dependency group

danceratopz committed:
chore(deps): add forgotten jinja2 dep to testing

danceratopz committed:
chore(deps): remove unrequired docs deps; rename `mkdocs`

danceratopz committed:
chore(tox): clean-up mkdocs related envs

danceratopz reviewed (commented):
Couple more observations.

danceratopz committed:
chore(ci): update name; auto-format yaml

danceratopz committed:
chore(ci): update name; auto-format yaml

danceratopz committed:
chore(ci): also run mkdocs build in test mkdocs

danceratopz committed:
fix(ci): fix glob pattern in markdownlint-cli2-action

danceratopz committed:
refactor(ci): use yaml anchor to avoid path repetition

danceratopz committed:
chore(docs): disable test case reference for tests/benchmarks cf #1756

danceratopz committed:
chore(docs): more robust eip checklist template path (cwd inepedendent)

danceratopz committed:
chore(tox): add all envs to envlist; add env descriptions

danceratopz committed:
refactor: remove use of string continuation via backslash

danceratopz committed:
style(tox): less confusing indentation

danceratopz commented:
@SamWilsn thanks for all of your review comments! I addressed them today and cleaned up a little. This would be good to from my side.

danceratopz reviewed (approved):
`pypy3` and `json_infra` failed due to 503 service unavailable for the ethquoakaops ubuntu package server (which has now resolved). Re-ran `pypy3`, will not re-run `json_infra`.
